onnxscript/onnx_opset/_impl/opset1.py

```diff
@@ -118,7 +122,9 @@ def Add(
             consumed_inputs=consumed_inputs,
         )
 
-    def And(self, A: BOOL, B: BOOL, axis: Optional[int] = None, broadcast: int = 0) -> BOOL:
+    T = TypeVar("T", bound=BOOL)
+
+    def And(self, A: T, B: T, axis: Optional[int] = None, broadcast: int = 0) -> BOOL:
```

Comment:
When there’s a single type, T and BOOL are equivalent and BOOL maybe simpler?

Reply:
I was on the fence about this case since allowing constraints with a single type to fall through means the constraint itself can't be resolved at eval time. It may not matter at all, but I'm inclined to keep single types as TypeVar for consistency.